script/script.go

```diff
@@ -59,35 +57,19 @@ func (s *Script) Remove(name string) bool {
 	return true
 }
 
-// SetBuiltinFunctions allows to define builtin functions.
-func (s *Script) SetBuiltinFunctions(funcs []*objects.BuiltinFunction) {
-	if funcs != nil {
-		s.builtinFuncs = make([]objects.Object, len(funcs))
-		for idx, fn := range funcs {
-			s.builtinFuncs[idx] = fn
-		}
-	} else {
-		s.builtinFuncs = []objects.Object{}
+// SetImports sets import modules.
+func (s *Script) SetImports(modules map[string]objects.Importable) {
+	if modules == nil {
+		s.importModules = map[string]objects.Importable{}
+		return
 	}
-}
 
-// SetBuiltinModules allows to define builtin modules.
-func (s *Script) SetBuiltinModules(modules map[string]*objects.ImmutableMap) {
-	if modules != nil {
-		s.builtinModules = make(map[string]objects.Object, len(modules))
-		for k, mod := range modules {
-			s.builtinModules[k] = mod
-		}
-	} else {
-		s.builtinModules = map[string]objects.Object{}
+	s.importModules = make(map[string]objects.Importable, len(modules))
+	for k, mod := range modules {
+		s.importModules[k] = mod
 	}
 }
```

Comment:
The entire function can be much simpler (modules is the same as s.importModules):

```
	s.importModules = modules

	if modules == nil {
		s.importModules = map[string]objects.Importable{}
	}
```

Reply:
Technically even leaving it nil is just fine as compiler.NewCompiler() deals with converting nil values to `map[string]objects.Importable{}`

Reply:
So the whole function can simply be:
```
	s.importModules = modules
```